hoot-core/src/main/cpp/hoot/core/elements/Way.cpp

```diff
@@ -285,6 +285,9 @@ void Way::removeNode(long id)
 
 void Way::replaceNode(long oldId, long newId)
 {
+  if (oldId == newId)
+    return;
```

Comment:
Sometimes a node is replaced by itself, no sense in running through the whole replacement function and updating indices if nothing actually changes.  Short circuit here.